### PR TITLE
refactor setup function

### PR DESCRIPTION
**Pull request by soundanalogous**
- split wifi init and firmata init into separate functions
- add hostConnectionCallback

### CODE REVIEW

File: examples/StandardFirmataWiFi/StandardFirmataWiFi.ino [hunk 1]
```diff
@@ -830,6 +830,28 @@ void systemResetCallback()
...
+ * Called when a TCP connection is either connected or disconnected.
+ * TODO - figure out why the callback is not being called when using ESP8266 as a TCP server and
+ * why only connect is called when using ESP8266 as a TCP client. In both cases the actual
+ * connection is working but not reported via the callback.
+ */
+void hostConnectionCallback(byte state)
+{
+  switch (state) {
+    case HOST_CONNECTION_CONNECTED:
+      DEBUG_PRINTLN( "TCP connection established" );
```
Comment (soundanalogous) [inline examples/StandardFirmataWiFi/StandardFirmataWiFi.ino]: TODO:
- Report connected or reconnected state to host 
- Report current state (per https://github.com/firmata/protocol/issues/59) if reconnect

**Comment by jnsbyr** [comment]
A change to the ```systemResetCallback()``` is required for the ESP8266, at least when SERIAL_DEBUG is enabled. A possible solution is described [here](https://github.com/firmata/arduino/pull/291#issuecomment-213808550).

**Comment by jnsbyr** [comment]
To prepare the usage of the hostConnectionCallback for the other host connection types where the transport layer is not connection oriented (serial, BLE serial) the watchog timer must be placed in a method that is periodically called (e.g. _maintain_). A possible solution is to insert a class inheritance level between Stream and all currently used connection type specific stream implementations, maybe calling it _FirmataSteam_. This class could handle 

- connect/reconnect distinction
- watchdog mode (disabled, send only, send and receive)
- watchdog sequence, telegram transmission and receive telegram monitoring
- calling the hostConnectionCallback for non connection oriented streams
- closing the connection oriented streams on watchdog timeout

**Comment by soundanalogous** [comment]
Going to merge this and defer any additional changes regarding hostConnectionCallback to a future PR after the protocol definitions for watchdog and connect/reconnect have been finalized.

**Pull request merged**